xlators/mgmt/glusterd/src/glusterd-volgen.c

```diff
@@ -2755,6 +2755,114 @@ check_and_add_debug_xl(volgen_graph_t *graph, dict_t *set_dict, char *volname,
     return ret;
 }
 
+gf_boolean_t
+check_user_xlator_position(dict_t *dict, char *key, data_t *value,
+                           void *prev_xlname)
+{
+    if (strncmp(key, "user.xlator.", strlen("user.xlator.")) != 0) {
+        return false;
+    }
+
+    if (fnmatch("user.xlator.*.*", key, 0) == 0) {
+        return false;
+    }
+
```

Comment:
We should make sure that the current `key` has format `user.xlator.*` to avoid problems later in `insert_user_xlator_to_graph`.

Reply:
@xhernandez Do you mean we should allow  even the dot after `user.xlator.$name`? The patch currently seems to allow `user.xlator.posix` but not `user.xlator.posix.one`

Reply:
If I understand the patch correctly, `user.xlator.<name>` will have the name of the previous xlator after which the user xlator must be loaded. `user.xlator.<name>.<option>` will be used to pass options to the xlator, and this is handled elsewhere. Here we are only searching for the first option, so I think it's correct.